### ISSUE TITLE
SkyImage error with FITS header COMMENT

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#701]
This is a bug in SkyImage FITS I/O related to the COMMENT in the FITS header reported by @leajouvin on Slack.

To reproduce read any FITS file with a multi-line comment, e.g.
```
>>> from gammapy.image import SkyImage
>>> image = SkyImage.read('test_datasets/unbundled/fermi/gll_iem_v02_cutout.fits')
```
Then when trying to write it, this error happens:
```
>>> image.write('temp.fits', clobber=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/gammapy-0.5.dev3368-py2.7-macosx-10.11-x86_64.egg/gammapy/image/core.py", line 262, in write
    hdu = self.to_image_hdu()
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/gammapy-0.5.dev3368-py2.7-macosx-10.11-x86_64.egg/gammapy/image/core.py", line 712, in to_image_hdu
    header.update(self.meta)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/astropy/io/fits/header.py", line 1155, in update
    update_from_dict(k, v)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/astropy/io/fits/header.py", line 1140, in update_from_dict
    card = Card(k, v)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/astropy/io/fits/card.py", line 448, in __init__
    self.value = value
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/astropy/io/fits/card.py", line 603, in value
    'represent printable characters in ASCII.' % value)
ValueError: FITS header values must contain standard printable ASCII characters; "  FITS (Flexible Image Transport System) format is defined in 'Astronomy\n  and Astrophysics', volume 376, page 359; bibcode: 2001A&A...376..359H" contains characters/bytes that do not represent printable characters in ASCII.
```
The problem is that the header contains a multi-line comment:
```
>>> from astropy.io.fits import getheader
>>> header = getheader('test_datasets/unbundled/fermi/gll_iem_v02_cutout.fits')
>>> print(repr(header))
...
EXTEND  =                    T / FITS dataset may contain extensions            
COMMENT   FITS (Flexible Image Transport System) format is defined in 'Astronomy
COMMENT   and Astrophysics', volume 376, page 359; bibcode: 2001A&A...376..359H 
FLUX    =        8.29632317174 /                                                
...
```
which is converted to a string with `\n`
```
>>> image.meta['COMMENT']
"  FITS (Flexible Image Transport System) format is defined in 'Astronomy\n  and Astrophysics', volume 376, page 359; bibcode: 2001A&A...376..359H"
```
 in `SkyImage.read` in this code
```python
        meta = OrderedDict(header)

        # parse astropy.io.fits.header._HeaderCommentaryCards as strings
        if 'HISTORY' in meta:
            meta['HISTORY'] = str(meta['HISTORY'])
        if 'COMMENT' in meta:
            meta['COMMENT'] = str(meta['COMMENT'])
```
from @adonath in https://github.com/gammapy/gammapy/commit/a3c7de339f69e7e89ea6266a65f26d5a5b5e1ef7 .

As a short-term fix, I'll comment out these lines of code in a separate PR.

I think (but didn't check) that `astropy.table.Table.read` also just doesn't do anything to process COMMENT in a special way in an attempt to preserve it. If so, maybe this short-term fix is also the long-term solution.

But I'm not sure and wan to discuss with @adonath .

Note that this issue only appears on Python 2, because for some reason (I presume) there `\n` is converted to a single non-ascii unicode character that's not allowed in FITS files.
Also testing this is a bit tricky (see #654).

**Comment by cdeil** [comment]
Fix is in PR #702.

**Comment by cdeil** [comment]
Actually, I'm pretty happy with the fix in #702 :
- Remove the problematic `COMMENT` and `HISTORY` keys from `image.meta` in `SkyImage.read`.
- Store the FITS header object as private `image._header`, because it's often useful for checking.
- Add a regression test.

So I'm closing this issue now and move on, because I think we should focus on features / docs / good tests for analysis, not these FITS header esoterics.

Of course, @adonath or anyone, if you want to change this, feel free to re-open this issue or a PR.
